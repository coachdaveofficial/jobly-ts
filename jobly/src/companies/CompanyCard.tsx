import React from "react";
import { Link } from "react-router-dom";
import { Company } from '../types/types';

export default function CompanyCard({ name, description, handle, logoUrl }: Company) {
  return (
    <Link
      className="CompanyCard card mb-4 bg-secondary text-light"
      to={`/companies/${handle}`}
      style={{ textDecoration: "none" }}
    >
      <div className="card-body">
        <h6 className="card-title">
          {name}
          {logoUrl && <img src={logoUrl}
            alt={name}
            className="float-right ml-5" />}
        </h6>
        <p ><small>{description}</small></p>
      </div>
    </Link>
  )
}