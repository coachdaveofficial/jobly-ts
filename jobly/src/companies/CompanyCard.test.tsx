import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyCard from "./CompanyCard";

it("matches the snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard handle={""} name={""} numEmployees={null} description={""} logoUrl={null} jobs={null} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function () {
    const { getByText } = render(
        <MemoryRouter>
            <CompanyCard handle={""} name={"testName"} numEmployees={null} description={"testDescription"} logoUrl={null} jobs={null} />
        </MemoryRouter>

    )

    expect(getByText("testName")).toBeInTheDocument();
    expect(getByText("testDescription")).toBeInTheDocument();

})