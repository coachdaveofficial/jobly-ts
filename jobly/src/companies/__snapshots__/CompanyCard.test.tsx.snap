// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard card"
    href="/companies/"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      />
      <p>
        <small />
      </p>
    </div>
  </a>
</DocumentFragment>
`;
